slobrok/src/vespa/slobrok/server/local_rpc_monitor_map.h

```diff
@@ -0,0 +1,61 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+#pragma once
+
+#include "map_listener.h"
+#include "map_source.h"
+#include "service_mapping.h"
+#include "service_map_history.h"
+#include "i_rpc_server_manager.h"
+#include "managed_rpc_server.h"
+#include "map_listener.h"
+#include "named_service.h"
+#include <vector>
+#include <memory>
+#include <map>
+
+namespace slobrok {
+
+/**
+ * @class LocalRpcMonitorMap
+ * @brief A collection of ManagedRpcServer objects
+ *
+ * Tracks up/down status for name->spec combinations
+ * that are considered for publication locally.
+ **/
+class LocalRpcMonitorMap : public IRpcServerManager,
+                           public MapListener
+{
+private:
+    struct PerService {
+        const ServiceMapping mapping;
+        bool up = false;
+        std::unique_ptr<ManagedRpcServer> srv = {};
+    };
+
+    using Map = std::map<vespalib::string, PerService>;
+
+    Map _map;
+    ServiceMapHistory _history;
+    FRT_Supervisor &_supervisor;
+    vespalib::GenCnt _genFromGlobal;
+
+    PerService * lookup(const ServiceMapping &mapping);
+    
+public:
+    LocalRpcMonitorMap(FRT_Supervisor &_supervisor);
+    ~LocalRpcMonitorMap();
+
+    ServiceMapHistory & localHistory();
```

Comment:
consider just calling this 'history', since the 'local' part relates to the class itself